mathics/eval/files_io/files.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+"""
+files-related evaluation functions
+"""
+
+from typing import Callable, Optional
+
+from mathics_scanner import TranslateError
+
+import mathics
+from mathics.core.builtin import MessageException
+from mathics.core.evaluation import Evaluation
+from mathics.core.parser import MathicsFileLineFeeder, parse
+from mathics.core.read import MathicsOpen
+from mathics.core.symbols import SymbolNull
+from mathics.core.systemsymbols import SymbolFailed, SymbolPath
+
+# Python representation of $InputFileName
+INPUT_VAR: str = ""
```

Comment:
I think that this should go inside the ``Definitions`` object.

Reply:
@mmatera I don't understand what you mean. Would you  please just go ahead and make the change you are suggesting? 

Also, I don't understand why in `set_inputfile()` we aren't storing `String` instead of `str` so we have to do String conversions on each evaluation of `$Input`.  Conceptually, in a definitions object we should prefer Mathics3 datatypes for session values that are user-accessible, right?

Reply:
As I mentioned before, IMHO, constants associated with the interpreter session should be stored in the ``Definitions`` object instead of using variables at the level of the modules. 

This allows, for example, two independent sessions of the Mathics interpreter in the same Python session. 

On the other hand, I agree that it would be better to store these variables as Mathics instead of Python native types to avoid unnecessary conversions.

In any case, both changes could be done in a subsequent PR

Reply:
Yes, I understood conceptually the idea. The details, I am not confident about and don't want or have the time to get into a back-and-forth: no that's not what I really meant, I meant, ... 

Ok. We'll merge this and feel free to iterate and go further.